src/tests/java/htsjdk/samtools/SAMUtilsTest.java

```diff
@@ -42,4 +44,35 @@ public void testCompareMapqs() {
         Assert.assertTrue(SAMUtils.compareMapqs(1, 255) > 0);
         Assert.assertTrue(SAMUtils.compareMapqs(2, 1) > 0);
     }
+
+    @Test
+    public void testClippingOfRecordWithSoftClipBasesAtTheEnd() {
+        /**
+         * Tests that if we need to clip a read with soft-clipping at the end, it does the right thing.
+         */
+
+        // setup the record
+        final SAMFileHeader header = new SAMFileHeader();
+        header.addSequence(new SAMSequenceRecord("1", 1000));
+        final SAMRecord record = new SAMRecord(header);
+        record.setReadPairedFlag(true);
+        record.setCigar(new Cigar(Arrays.asList(new CigarElement(5, CigarOperator.M), new CigarElement(5, CigarOperator.S)))); // 5M5S
+        record.setReferenceIndex(0);
+        record.setAlignmentStart(1);
+        record.setMateAlignmentStart(5); // should overlap by one base now
+        record.setMateReferenceIndex(0);
+        record.setReadBases("AAAAAAAAAA".getBytes());
+
+        final int numToClip = SAMUtils.getNumOverlappingBasesToClip(record);
+        Assert.assertEquals(numToClip, 1);
+
+        SAMUtils.clipOverlappingBases(record, numToClip, false); // Side-effects are OK
+
+        final Cigar cigar = record.getCigar();
+        Assert.assertEquals(cigar.numCigarElements(), 2);
+        Assert.assertEquals(cigar.getCigarElement(0).getLength(), 4); // 4M
```

Comment:
Any reason not to compare the cigars directly? Or the cigar-strings if that doesn't work?